### ISSUE TITLE
F-Droid can't build

### ISSUE THREAD

**Issue by licaon-kter** [ente-io/photos-app#1363]
Just seeing https://gitlab.com/fdroid/fdroiddata/-/jobs/4974953268#L7301 which means you've forgot to bump Flutter, amirite? Oh, look sooo old: https://github.com/ente-io/photos-app/tree/v0.7.87/thirdparty :)

But I can force it to 3.13.1 for kicks... and then... fails later with
```
...
Running Gradle task 'assembleFdroidRelease'...                  
...
This will generate a JSON format file containing all messages that 
need to be translated.
.pub-cache/hosted/pub.dev/extended_image-8.0.2/lib/src/extended_image.dart:976:36: Error: The argument type 'BorderRadius?' can't be assigned to the parameter type 'BorderRadiusGeometry' because 'BorderRadius?' is nullable and 'BorderRadiusGeometry' isn't.
 - 'BorderRadius' is from 'package:flutter/src/painting/border_radius.dart' ('thirdparty/flutter/packages/flutter/lib/src/painting/border_radius.dart').
 - 'BorderRadiusGeometry' is from 'package:flutter/src/painting/border_radius.dart' ('thirdparty/flutter/packages/flutter/lib/src/painting/border_radius.dart').
              borderRadius: widget.borderRadius,
                                   ^
.pub-cache/hosted/pub.dev/extended_image-8.0.2/lib/src/image/painting.dart:202:13: Error: 'ImageTilingInfo' isn't a type.
      final ImageTilingInfo info =
            ^^^^^^^^^^^^^^^
.pub-cache/hosted/pub.dev/extended_image-8.0.2/lib/src/image/painting.dart:203:11: Error: Method not found: 'createTilingInfo'.
          createTilingInfo(repeat, rect, destinationRect, sourceRect);
          ^^^^^^^^^^^^^^^^
lib/ui/lifecycle_event_handler.dart:15:13: Error: The type 'AppLifecycleState' is not exhaustively matched by the switch cases since it doesn't match 'AppLifecycleState.hidden'.
 - 'AppLifecycleState' is from 'dart:ui'.
Try adding a default case or cases that match 'AppLifecycleState.hidden'.
    switch (state) {
            ^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.

* Where:
Script '/home/vagrant/build/io.ente.photos.fdroid/thirdparty/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1297

* What went wrong:
Execution failed for task ':app:compileFlutterBuildFdroidRelease'.
> Process 'command '/home/vagrant/build/io.ente.photos.fdroid/thirdparty/flutter/bin/flutter'' finished with non-zero exit value 1
```

Mmmkay, so something is wrong... 

Actual full log [io.ente.photos.fdroid_487.log.gz](https://github.com/ente-io/photos-app/files/12473877/io.ente.photos.fdroid_487.log.gz)

What did I miss? Something was "removed" and needs to be rebuilt?

/LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/9ad54b68284e67dd0b455122fb61334bc9ecdf45

**Comment by ua741** [comment]
Thank you for flagging this issue. I vaguely remember updating the submodule, but clearly that doesn't appear to be the case. We will publish a new tag after including a couple of bug fixes in the upcoming release.

**Comment by licaon-kter** [comment]
Thanks, that fixes it

**Comment by senenpalanca97** [comment]
Hi! Same problem here. What u did to fix it? Is a problem with extended_image compatibility?

**Comment by ua741** [comment]
> Hi! Same problem here. What u did to fix it? Is a problem with extended_image compatibility?

I am assuming that you are running into this issue for another app.
If you have recently upgraded flutter version, then you also need to upgrade `extended_image`. 

Fwiw, to mitigate the risk of security vulnerability: [CVE-2023-4863 ](https://nvd.nist.gov/vuln/detail/CVE-2023-4863), we recently [upgraded](https://github.com/ente-io/photos-app/pull/1421) the flutter version for our app

**Comment by licaon-kter** [comment]
https://github.com/ente-io/photos-app/blob/fdroid-v0.7.99/.github/workflows/release.yml#L29 -> 3.13.4 :)

https://github.com/ente-io/photos-app/tree/fdroid-v0.7.99/thirdparty -> 3.10.6 :(

**Comment by ua741** [comment]
Patched in [fdroid-v0.7.100](https://github.com/ente-io/photos-app/tree/fdroid-v0.7.100/thirdparty). 
Apologies for breaking the build again. We updated the version in the main branch, and the merge was clean. I'm not sure why the f-droid branch was still pointing to the older version. I'll investigate this during the next version upgrade.
